src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -113,6 +113,9 @@
     // A set of methods that result in dialog creations.
     protected static final Set<String> dialogCreatingMethods = new HashSet<String>();
 
+    private transient Set<SIPDialogInjectionListener> dialogInjectionListeners = 
```

Comment:
An ArrayList would have the behavior (view other comment). I'm not sure what "naturally ordered" means in context with ConcurrentSkipList